bundles/org.openhab.core.model.script/src/org/openhab/core/model/script/lib/NumberExtensions.java

```diff
@@ -380,6 +380,24 @@ public static BigDecimal numberToBigDecimal(Number number) {
             }
             return null;
         }
+        if (number instanceof DecimalType dtype) {
+            return dtype.toBigDecimal();
+        }
+        if (number instanceof BigDecimal decimal) {
+            return decimal;
+        }
+        if (number instanceof Double value) {
+            return new BigDecimal(value);
```

Comment:
You are right it is better to run into the default case here, so i will remove the Double and Float case here.
I think the impact is not that big, this will only hit when a hardcoded value is applied and this should be without a crazy long fraction part.